packages/flutter/lib/src/material/time.dart

```diff
@@ -0,0 +1,187 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' show hashValues;
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+import 'material_localizations.dart';
+
+/// Whether the [TimeOfDay] is before or after noon.
+enum DayPeriod {
+  /// Ante meridiem (before noon).
+  am,
+
+  /// Post meridiem (after noon).
+  pm,
+}
+
+/// A value representing a time during the day.
+@immutable
+class TimeOfDay {
```

Comment:
Assuming that this class isn't exclusively for the material time picker, we should consider moving it and the corresponding material localization code, to the widgets library and the WidgetsLocalizations class.